src/components/Features/Editor/CodeEditor/CodeEditor.js

```diff
@@ -0,0 +1,85 @@
+import CodeMirror from 'codemirror';
+import 'codemirror/addon/mode/simple';
+import 'codemirror/keymap/sublime';
+import 'codemirror/lib/codemirror.css';
+import 'codemirror/theme/ambiance.css';
+import PropTypes from 'prop-types';
+import React, {forwardRef} from 'react';
+import {Controlled as CodeMirror2} from 'react-codemirror2';
+import {AppModes} from '../../../../constants/app-modes';
+import styles from './CodeEditor.module.scss';
+
+CodeMirror.defineSimpleMode(AppModes.CAIRO, {
+  start: [
+    {
+      regex:
+        /(?:func|struct|namespace|end|call|ret|jmp|if|let|const|import|from|as|abs|rel|local|static_assert|tempvar|return|assert|member|felt|cast|else|alloc_locals|with|nondet)\b/,
+      token: 'keyword'
+    },
+    {regex: /ap|fp/, token: 'atom'},
+    {
+      regex: /-?[0-9]+/,
+      token: 'number'
+    },
+    {regex: /#.*/, token: 'comment'},
+    {regex: /[-+*=<>!]+/, token: 'operator'},
```

Comment:
doesn't this regex mean that it will paint also strings like '!!' and '<<>' as operators?

Reply:
Not really sure, this is done by @liorgold2 to support cairo-lang and I've pasted it as-is from the previous version.

Reply:
But for your Q regarding the regex itself, yes, you're right. 
Again, not sure why it has been done that way.

Reply:
I suspect it's only for syntax highlighting.
so potentially, if someone write something like `a <<+ b` it will show the `<<+` as an operator, but the compiler will fail it.